src/native/minipal/time.c

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <assert.h>
+#include <minipal/time.h>
+
+#ifdef HOST_WINDOWS
+
+#include <Windows.h>
+
+int64_t minipal_hires_ticks()
+{
+    LARGE_INTEGER ts;
+    QueryPerformanceCounter(&ts);
+    return ts.QuadPart;
+}
+
+int64_t minipal_hires_tick_frequency()
+{
+    LARGE_INTEGER ts;
+    QueryPerformanceFrequency(&ts);
+    return ts.QuadPart;
+}
+
+#else // HOST_WINDOWS
+
+#include "minipalconfig.h"
+
+#include <time.h> // nanosleep
+#include <errno.h>
+
+inline void YieldProcessor()
+{
+#if defined(HOST_X86) || defined(HOST_AMD64)
+    __asm__ __volatile__(
+        "rep\n"
+        "nop");
+#elif defined(HOST_ARM)
+    __asm__ __volatile__( "yield");
+#elif defined(HOST_ARM64)
+    __asm__ __volatile__(
+        "dmb ishst\n"
+        "yield"
+        );
+#elif defined(HOST_LOONGARCH64)
+    __asm__ volatile( "dbar 0;  \n");
+#elif defined(HOST_RISCV64)
+    // TODO-RISCV64-CQ: When Zihintpause is supported, replace with `pause` instruction.
+    __asm__ __volatile__(".word 0x0100000f");
+#else
+    return;
+#endif
+}
+
+#define tccSecondsToNanoSeconds 1000000000      // 10^9
+int64_t minipal_hires_tick_frequency()
+{
+    return tccSecondsToNanoSeconds;
+}
+
+int64_t minipal_hires_ticks()
+{
+#if HAVE_CLOCK_GETTIME_NSEC_NP
+    return (int64_t)clock_gettime_nsec_np(CLOCK_UPTIME_RAW);
+#else
+    struct timespec ts;
+    int result = clock_gettime(CLOCK_MONOTONIC, &ts);
+    if (result != 0)
+    {
+        assert(!"clock_gettime(CLOCK_MONOTONIC) failed");
+        __builtin_unreachable();
+    }
+
+    return ((int64_t)(ts.tv_sec) * (int64_t)(tccSecondsToNanoSeconds)) + (int64_t)(ts.tv_nsec);
+#endif
+}
+
+#endif // !HOST_WINDOWS
+
+void minipal_microsleep(uint32_t usecs, uint32_t* usecsSinceYield)
+{
+#ifdef HOST_WINDOWS
+    if (usecs > 1000)
+    {
+        SleepEx(usecs / 1000, FALSE);
+        if (usecsSinceYield)
+        {
+            usecsSinceYield = 0;
+        }
+
+        return;
+    }
+#else
+    if (usecs > 10)
```

Comment:
This is the actual fix. When we need to pause for longer than 10 microseconds, we sleep instead, thus freeing compute resources to other threads (including ones that we want to suspend).

Reply:
`10` is a somewhat arbitrary number in the order of a context switch cost. (2-10 microseconds). 
We are not very sensitive to the exact value. It just means "if pausing for less, might be ok to spin-wait".

Reply:
The other reason for `10` is that the default timer slack on Linux is 50 microseconds. So the shortest sleep could be on average 25 microseconds anyways if our wakeups are bunched up with other sleepers.
 
Thus `10` seems like a reasonable threshold to use sleep.